src/__tests__/lib/containers/UserCardTests.test.tsx

```diff
@@ -172,29 +175,26 @@ describe('it creates the correct UI for the small card', () => {
     ).toThrowError()
 
     // Hover over the username
-    await act(async () => {
-      userEvent.hover(screen.getByText(`@${props.userProfile.userName}`))
-
-      // The card should appear, which would let us see first/last name
-      // we have to wrap in a waitFor because of the delay
-      await waitFor(() =>
-        expect(() =>
-          screen.getAllByText(
-            `${props.userProfile.firstName} ${props.userProfile.lastName}`,
-          ),
-        ).not.toThrowError(),
-      )
-
-      // Unhover and confirm that the card disappears (we will no longer see a full name anywhere)
-      userEvent.unhover(screen.getByText(`@${props.userProfile.userName}`))
-      await waitFor(() =>
-        expect(() =>
-          screen.getAllByText(
-            `${props.userProfile.firstName} ${props.userProfile.lastName}`,
-          ),
-        ).toThrowError(),
-      )
-    })
+    userEvent.hover(screen.getByText(`@${props.userProfile.userName}`))
+
+    // The card should appear, which would let us see first/last name
+    // we have to wrap in a waitFor because of the delay
+    await waitFor(() =>
+      screen.getAllByText(
+        `${props.userProfile.firstName} ${props.userProfile.lastName}`,
+      ),
+    )
+    await waitFor(() => screen.getByText('ORCID', { exact: false }))
```

Comment:
This fixes the `not wrapped in act(...)` warning, but I realized we see this warning so much because we don't have a mechanism for cancelling our web requests in `useEffect` when we tear down a component. Should we look into doing this with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort) or [axios](https://axios-http.com/docs/cancellation)?

Reply:
Do you mean cancelling the calls done by SynapseClient (which use fetch)?

Reply:
Yes